tests/ContaoCommunityAlliance/DcGeneral/Test/DataDefinition/Definition/DefaultPropertiesDefinitionTest.php

```diff
@@ -0,0 +1,283 @@
+ *
+ * @coversDefaultClass \ContaoCommunityAlliance\DcGeneral\DataDefinition\Definition\DefaultPropertiesDefinition
+ */
+class DefaultPropertiesDefinitionTest extends TestCase
+{
+    /**
+     * Get the default properties definition.
+     *
+     * @return DefaultPropertiesDefinition
+     */
+    private function getDefintiion()
+    {
+        return new DefaultPropertiesDefinition();
+    }
+
+    /**
+     * Get a default property.
+     *
+     * @param string $name The property name.
+     *
+     * @return DefaultProperty
+     */
+    private function getDefaultProperty($name)
+    {
+        return new DefaultProperty($name);
+    }
+
+    /**
+     * Get a default collection property.
+     *
+     * @param string $name The property name.
+     *
+     * @return DefaultSubProperty
+     */
+    private function getDefaultCollectionProperty($name)
+    {
+        return new DefaultSubProperty($name);
+    }
+
+    /**
+     * Get a default propert with collectiony.
+     *
+     * @param string $name The property name.
+     *
+     * @return DefaultCompoundProperty
+     */
+    private function getDefaultPropertyWithCollection($name)
+    {
+        return new DefaultCompoundProperty($name);
+    }
+
+    /**
+     * Test add default property.
+     *
+     * @covers ::addProperty
+     * @covers ::getProperties
+     * @covers ::hasProperty
+     */
+    public function testAddDefaultProperty()
+    {
+        $definition = $this->getDefintiion();
+        $property   = $this->getDefaultProperty('dummy-property');
+
+        $this->assertInternalType('array', $definition->getProperties());
+        $this->assertFalse($definition->hasProperty($property->getName()));
+        $this->assertFalse($definition->hasProperty('dummy-property-foo'));
+
+        $definition->addProperty($property);
+
+        $this->assertInternalType('array', $definition->getProperties());
+        $this->assertTrue($definition->hasProperty($property->getName()));
+        $this->assertFalse($definition->hasProperty('dummy-property-foo'));
+    }
+
+    /**
+     * Test exceptions for method add properties.
+     *
+     * @expectedException \ContaoCommunityAlliance\DcGeneral\Exception\DcGeneralInvalidArgumentException
+     *
+     * @covers ::addProperty
+     */
+    public function testAddPropertyExceptionInvalidInstance()
+    {
+        $definition = $this->getDefintiion();
+        $property   = $this->getDefaultProperty('dummy-property');
+
+        $definition->addProperty($property);
+
+        $this->setExpectedException(
+            'ContaoCommunityAlliance\DcGeneral\Exception\DcGeneralInvalidArgumentException',
+            'Passed value is not an instance of PropertyInterface.'
+        );
+        $definition->addProperty(array());
+    }
+
+    /**
+     * Test exceptions for method add property.
+     *
+     * @expectedException \ContaoCommunityAlliance\DcGeneral\Exception\DcGeneralInvalidArgumentException
```

Comment:
Annotation can be removed due to $this->setExpectedException below

Reply:
change